### PR TITLE
Rename extension hibernate-search-elasticsearch to hibernate-search-orm-elasticsearch

### PR DESCRIPTION
**Pull request by yrodiere**
@gsmet As discussed together.

There's been talks about introducing another Hibernate Search extension that targets Elasticsearch, but is not tied to Hibernate ORM, allowing the use of Hibernate Search to replicate data from arbitrary datastores (MongoDB, ...).

When that happens (no schedule at the moment), we will end up with an extension named `hibernate-search-elasticsearch` that bundles Hibernate Search, targets Elasticsearch, but *cannot* be used by MongoDB users, who must rely on another extension, probably named `hibernate-search-standalone-elasticsearch`.

Even worse, the configuration prefix `quarkus.hibernate-search.` will be dedicated to Hibernate Search configuration within Hibernate ORM, any other extensions will have to use a different prefix such as `quarkus.hibernate-search-standalone.`.

This will be confusing.

By inserting "orm" into the name of the Hibernate Search ORM+Elasticsearch extension, I intend to clarify that this extension is about Hibernate ORM.

We should add this to the migration guide:

``````
## Hibernate Search ORM + Elasticsearch (Preview)

* You should update your Maven/Gradle dependencies: replace any occurrence of the artifact ID `hibernate-search-elasticsearch` with `hibernate-search-orm-elasticsearch`
* You should update your configuration: replace any occurrence of the prefix `quarkus.hibernate-search.` with `quarkus.hibernate-search-orm.`
``````

### CODE REVIEW

File: extensions/hibernate-search-orm-elasticsearch/deployment/pom.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <modelVersion>4.0.0</modelVersion>
 
     <artifactId>quarkus-hibernate-search-orm-elasticsearch-deployment</artifactId>
-    <name>Quarkus - Hibernate Search - Elasticsearch - Deployment</name>
+    <name>Quarkus - Hibernate Search - ORM+Elasticsearch - Deployment</name>
```
Comment (gsmet) [inline extensions/hibernate-search-orm-elasticsearch/deployment/pom.xml:14]: I would have spaces around the `+`?

**Review by gsmet — APPROVED** [review]
Looks good except for a small question.

**Comment by yrodiere** [comment]
I addressed your comment. Thanks.

**Comment by gsmet** [comment]
Migration guide updated.

**Comment by rsvoboda** [comment]
This change "broke" my application. It's nice to have entry in https://github.com/quarkusio/quarkus/wiki/Migration-Guide-1.10 but I think there should be more fuzz around it. Added `release/breaking-change` label

**Pull request merged**